spec/unit/pdk/validate/puppet/puppet_environment_conf_validator_spec.rb

```diff
@@ -0,0 +1,158 @@
+
+      it { is_expected.to eq(false) }
+    end
+
+    context 'in an unknown context' do
+      let(:validator_context) { PDK::Context::None.new(nil) }
+
+      it { is_expected.to eq(false) }
+    end
+  end
+
+  describe '.validate_target' do
+    subject(:return_value) { described_class.new.validate_target(report, target[:name]) }
+
+    let(:report) { PDK::Report.new }
+
+    before(:each) do
+      [
+        target[:name],
+        File.join(validator_context.root_path, target[:name]),
+      ].each do |filename|
+        allow(PDK::Util::Filesystem).to receive(:directory?).with(filename).and_return(target.fetch(:directory, false))
+        allow(PDK::Util::Filesystem).to receive(:file?).with(filename).and_return(target.fetch(:file, true))
+        allow(PDK::Util::Filesystem).to receive(:readable?).with(filename).and_return(target.fetch(:readable, true))
+        allow(PDK::Util::Filesystem).to receive(:read_file).with(filename).and_return(target.fetch(:content, ''))
+      end
+    end
+
+    context 'when a target is provided that is an unreadable file' do
+      let(:target) { { name: 'environment.conf', readable: false } }
+
+      it 'adds a failure event to the report' do
+        expect(report).to receive(:add_event).with(
+          file:     target[:name],
+          source:   validator.name,
+          state:    :failure,
+          severity: 'error',
+          message:  'Could not be read.',
+        )
+        expect(return_value).to eq(1)
+      end
+    end
+
+    context 'when a target is provided that is not a file' do
+      let(:target) { { name: 'environment.conf', file: false } }
+
+      it 'skips the target' do
+        expect(report).not_to receive(:add_event)
+      end
+    end
+
+    context 'when a target is provided that is empty' do
+      let(:target) { { name: 'environment.conf', content: '' } }
+
+      it 'adds a passing event to the report' do
+        expect(report).to receive(:add_event).with(
+          file:     target[:name],
+          source:   validator.name,
+          state:    :passed,
+          severity: 'ok',
+        )
+        expect(return_value).to eq(0)
+      end
+    end
+
+    context 'when a target is provided that has all allowed settings and is valid' do
+      let(:target) { { name: 'environment.conf', content: "modulepath=foo\nmanifest=foo\nconfig_version=foo\nenvironment_timeout=0" } }
+
+      it 'adds a passing event to the report' do
+        expect(report).to receive(:add_event).with(
+          file:     target[:name],
+          source:   validator.name,
+          state:    :passed,
+          severity: 'ok',
+        )
+        expect(return_value).to eq(0)
+      end
+    end
+
+    context 'when a target is provided that has invalid settings and sections' do
+      let(:target) { { name: 'environment.conf', content: "modulepath=foo\nmanifest=foo\nconfig_version=foo\nenvironment_timeout=foo\ninvalid=true\n[invalid_section]\ninvalid=true" } }
```

Comment:
Ditto: "Maybe move `content` to it's own var/`let` block and use a heredoc to make it easier to maintain?"

Reply:
See above.